modules/era/lua_dynamis/globals/era_dynamis.lua

```diff
@@ -971,7 +988,13 @@ xi.dynamis.handleDynamis = function(zone)
 
     if zone:getLocalVar(string.format("[DYNA]NoPlayersInZone_%s", zoneID)) ~= 0 then
         if zone:getLocalVar(string.format("[DYNA]NoPlayersInZone_%s", zoneID)) <= os.time() then -- If cooldown period eclipses current OS time, cleanup.
-            xi.dynamis.cleanupDynamis(zone)
+            if
+                GetServerVariable(string.format("[DYNA]ZoneCooldown_%s", xi.dynamis.dynaIDLookup[zoneID].entryZone)) == 0 -- if no cleanup timer is set
+            then
+                SetServerVariable(string.format("[DYNA]ZoneCooldown_%s", xi.dynamis.dynaIDLookup[zoneID].entryZone), os.time() + 90) -- Set a 90s timer for cleanup
+            end
```

Comment:
When the times is 0 isn't this going to trigger every time always adding 90 seconds to the timer? Or am I not understanding this correctly.

Reply:
The logic works as so:
The timer is only set to 0 when a new dynamis is started/hourglass traded (Timer is per starting zone ID [ie: S.Sandy/BastokMarkets/etc]).
When the zone expires, or when it remains empty for at least 15 minutes, then the timer is initiated with os.time() + 90s. (This is only going to happen when all other checks have passed and the zone is going to run its cleanup script).
Once the 90s additional cleanup time has passed and a new player trades a glass, that timer var is then "reset" back to 0.

So the setting/clearing of the 90s cooldown only happens during zone expiration or zone initialization.

Because the timer is only set to 0 upon trading of an hourglass, the old time will remain in place.
Side Note: This still works even after a restart/zone reload because the timer will initially be set to 0 and os.time() will always be greater than 0, meaning that there is no timer for the first trade after the zone/server is reset.

I can upload a short video with debugs if you need it.